File: src/main/java/com/ririv/quickoutline/view/controls/SvgIcon.java
package com.ririv.quickoutline.view.controls;

import com.ririv.quickoutline.view.utils.BatikToJavaFXConverter;
import javafx.scene.layout.Background;
import javafx.scene.layout.Region;
import javafx.scene.paint.Color;

/**
 * A base class for creating icons from SVG files using -fx-shape.
 */
public class SvgIcon extends Region {

    /**
     * Sets the SVG path for this icon.
     * @param svgPath The resource path to the SVG file.
     */
    protected void setSvgPath(String svgPath) {
        String combinedPath = BatikToJavaFXConverter.getCombinedPath(svgPath);
        if (!combinedPath.isEmpty()) {
            this.setStyle("-fx-shape: \"" + combinedPath + "\";");
        }
        // Add a default style class for all icons
        this.getStyleClass().add("icon");

        setBackground(Background.fill(Color.GRAY));
    }
}

File: src/main/kotlin/com/ririv/quickoutline/view/icons/SvgIcon.kt
package com.ririv.quickoutline.view.icons

import androidx.compose.material3.Icon
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import com.ririv.quickoutline.view.ui.loadResourcePainter

@Composable
fun SvgIcon(
    resource: String,
    modifier: Modifier = Modifier,
    tint: Color = Color.Unspecified
) {
    Icon(
        painter = loadResourcePainter(resource),
        contentDescription = null,
        modifier = modifier,
        tint = tint
    )
}
